### PR TITLE
Replace random non-ascii-print characters with standard substitutes

### PR DESCRIPTION
**Pull request by ptr727**
# What does this implement/fix?

Replace non-ascii-print characters with ascii equivalents, unless explicitly required for functionality.  
Discovered via VSCode setting for invisible characters, regex find in files `[^\x00-\x7f]`, Gremlins Tracker extension `
nhoizey.gremlins`

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by ssieb** [comment]
Why is it draft?

File: esphome/components/pmwcs3/sensor.py [hunk 1]
```diff
@@ -62,7 +62,7 @@
             cv.Optional(CONF_VWC): sensor.sensor_schema(
                 icon=ICON_ALPHA,
                 accuracy_decimals=3,
-                unit_of_measurement="cm3cm−3",
+                unit_of_measurement="cm3cm-3",
```
Comment (jesserockz) [inline esphome/components/pmwcs3/sensor.py]: This change might need testing to ensure it doesn't break statistics

Comment (ptr727) [reply esphome/components/pmwcs3/sensor.py]: Hmm, yes, not sure how HA and SQL will handle this. Could no find anything about specific about SQL codepage conversions in HA, so better to treat it like other explicit units, although the choice of `−` vs. `-` was unfortunate.

**Comment by ptr727** [comment]
> Why is it draft?

Wanted to test the appetite for such a nitpicking PR, submitted.

File: esphome/components/max6956/max6956.h
Comment (looping40) [inline esphome/components/max6956/max6956.h:1]: The changes are only in comments text.

**Review by clydebarrow — APPROVED** [review]
LGTM

**Pull request merged**